pkg/vcMock/mock.go

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Description: A mock implementation of virtcontainers that can be used
+// for testing.
+//
+// This implementation calls the function set in the object that
+// corresponds to the name of the method (for example, when CreatePod()
+// is called, that method will try to call CreatePodFunc). If no
+// function is defined for the method, it will return an error in a
+// well-known format. Callers can detect this scenario by calling
+// IsMockError().
+
+package vcMock
+
+import (
+	"fmt"
+	"syscall"
+
+	"github.com/Sirupsen/logrus"
+	vc "github.com/containers/virtcontainers"
+)
+
+// mockErrorPrefix is a string that all errors returned by the mock
+// implementation itself will contain as a prefix.
+const mockErrorPrefix = "vcMock forced failure"
+
+// SetLogger implements the VC function of the same name.
+func (m *VCMock) SetLogger(logger logrus.FieldLogger) {
+	if m.SetLoggerFunc != nil {
+		m.SetLoggerFunc(logger)
+		return
+	}
+}
+
+// CreatePod implements the VC function of the same name.
+func (m *VCMock) CreatePod(podConfig vc.PodConfig) (vc.VCPod, error) {
+	if m.CreatePodFunc != nil {
```

Comment:
Maybe I am missing something. Where are these functions (*Func) getting assigned? Or are they supposed to always fail unless they are explicitly assigned in the test?